supports scores up to 2 billion

unbibium opened the pull request:
Pico-8's Lua allows limited use of the entire 32 bits of the number type as an integer.  so I created a bigscore object that handles addition, multiplication, and one kind of comparison.  I use this type for both the score and the goal score.

It will probably only work up to about 2 billion, so I capped the score there.

unbibium committed:
supports scores up to 2 billion

angelvalentin80 commented:
![scoring-bug](https://github.com/user-attachments/assets/53b5ee6f-8e31-4dfb-9d5d-287f05ff1d76)

Hey @unbibium

I tested out the code and it seems like the score is going over the cap for the "Next Goal" I also did score some hands that went over the previous cap

But for some reason I got this error where I should of had 40k score, but it ended up showing "Scored Last" as 2billion

angelvalentin80 reviewed (changes requested):
It seems as though I got this bug. 
The score on the right is what I should of "Scored Last" but now it's showing as 2 billion. I wonder what may have happened there?

![scoring-bug](https://github.com/user-attachments/assets/2d23e2c5-9b3f-4d7b-acd1-1d956c62a484)

angelvalentin80 commented on balatro-demo.p8:
```diff
@@ -665,10 +717,8 @@ end
-	if goal_score < 0 then
-		goal_score = score_cap
-	end
+	goal_score *= 1.5
```
I am trying it locally, and I think I fixed it by changing this line to

goal_score = goal_score * 1.5

I prefer the goal_score *= 1.5 syntax, but earlier on in the project that syntax was causing problem for whatever reason. I don't know if it's a PICO8 thing or what. SO I just decided to start doing x = x * y instead of x *= y

![100k](https://github.com/user-attachments/assets/0347379a-44ee-46a5-aed0-ecaef29a43cc)

angelvalentin80 replied:
ANDD after more testing I came across the bug again

angelvalentin80 replied:
I really am not sure what the issue could be

unbibium replied:
i added some assertions to make sure it's behaving the way I hope for cases I could think of.  if you find a case that doesn't work then you can add it there.

also, even though it made the methods bigger, I added overflow detection to `__add` and `__mul` directly so that if it detects an overflow, it'll return an object that does nothing but display "NANEINF"

unbibium committed:
bigscore returns NANeINF if overflow

probably not foolproof for all cases, but there are
unit tests so if you find a case that doesn't work,
add it there.

unbibium commented:
my last commit also includes the goal_score syntax change just in case
